spec/integration/spec_helper.cr

```diff
@@ -0,0 +1,172 @@
+ENV["PATH"] = "#{File.expand_path("../../bin", __DIR__)}:#{ENV["PATH"]}"
+ENV["SHARDS_CACHE_PATH"] = ".shards"
+
+require "spec"
+require "../../src/config"
+require "../../src/lock"
+require "../../src/spec"
+require "../support/factories"
+require "../support/cli"
+
+module Shards::Specs
+  @@created_repositories = false
+
+  def self.created_repositories?
+    @@created_repositories
+  end
+
+  def self.created_repositories!
+    @@created_repositories = true
+  end
+end
+
+Spec.before_each do
```

Comment:
Why does this run before each when there's a guard to only executed it once? Can't we just run the setup directly without a hook?

Reply:
There is no `Spec.before_all` 🤔 . But we can remove that initialization check and move it to the top level.

Reply:
There is... https://crystal-lang.org/api/master/Spec/Methods.html#before_all(&block)-instance-method

Reply:
That method only works inside a `describe` block. This seems a bit inconsistent, though. We should probably consider improving this API and maybe also add a top-level `before_all`.

Reply:
I just realized there is actually `Spec.before_suite`. And even we core members were not aware it exists :D

Reply:
Done before_suite is used.